pkg/block/s3/adapter.go

```diff
@@ -870,12 +870,12 @@ func (a *Adapter) ResolveNamespace(storageNamespace, key string, identifierType
 }
 
 func (a *Adapter) GetRegion(ctx context.Context, storageNamespace string) (string, error) {
-	bucket, found := strings.CutPrefix(storageNamespace, fmt.Sprintf("%s://", block.BlockstoreTypeS3))
-	if !found {
-		return "", fmt.Errorf(`%w: "s3://" prefix not found %s`, block.ErrInvalidNamespace, storageNamespace)
+	namespaceUrl, err := url.Parse(storageNamespace)
+	if err != nil {
+		return "", fmt.Errorf(`%w: %s isn't a valid url'`, block.ErrInvalidNamespace, storageNamespace)
```

Comment:
The current convention for err messages
```suggestion
		return "", fmt.Errorf(`%s isn't a valid url': %w`, storageNamespace, block.ErrInvalidNamespace)
```